### PR TITLE
Buddy.works support

### PR DESCRIPTION
**Pull request by wbeuil**
Add https://buddy.works/ support

### CODE REVIEW

**Comment by drazisil** [comment]
@thomasrockhu

File: tests/env [hunk 1]
```diff
@@ -121,3 +121,11 @@ export CODEBUILD_RESOLVED_SOURCE_VERSION=""
...
 export CODEBUILD_BUILD_ID=""
 export CODEBUILD_SOURCE_REPO_URL=""
+export BUDDY=""
+export BUDDY_EXECUTION_BRANCH=""
+export BUDDY_EXECUTION_REVISION=""
+export BUDDY_EXECUTION_ID=""
+export BUDDY_EXECUTION_URL=""
+export BUDDY_EXECUTION_PULL_REQUEST_NO=""
+export BUDDY_EXECUTION_TAG=""
+export BUDDY_REPO_SLUG=""
```
Comment (thomasrockhu) [inline tests/env:131]:
```suggestion
export BUDDY_REPO_SLUG=""
```

Comment (thomasrockhu) [inline tests/env:131]:
```suggestion
export BUDDY_REPO_SLUG=""
```

File: codecov [hunk 2]
```diff
@@ -816,6 +816,20 @@ then
   then
     commit=$(git rev-parse "$BITBUCKET_COMMIT")
   fi
+
+elif [ "$CI" = "true" ] && [ "$BUDDY" = "true" ];
```
Comment (thomasrockhu) [inline codecov:820]:
```suggestion
elif [ "$CI" = "true" ] && [ "$BUDDY" = "true" ];
```

File: codecov [hunk 3]
```diff
@@ -816,6 +816,20 @@ then
...
+  # https://buddy.works/docs/pipelines/environment-variables
+  service="buddy"
+  branch="$BUDDY_EXECUTION_BRANCH"
+  build="$BUDDY_EXECUTION_ID"
+  build_url=$(urlencode "$BUDDY_EXECUTION_URL")
+  commit="$BUDDY_EXECUTION_REVISION"
+  pr="$BUDDY_EXECUTION_PULL_REQUEST_NO"
+  tag="$BUDDY_EXECUTION_TAG"
+  slug="$BUDDY_REPO_SLUG"
+
```
Comment (thomasrockhu) [inline codecov:832]:
```suggestion
  slug="$BUDDY_REPO_SLUG"
```

File: codecov [hunk 4]
```diff
@@ -827,6 +841,7 @@ then
   commit="$CIRRUS_CHANGE_IN_REPO"
   build="$CIRRUS_TASK_ID"
   job="$CIRRUS_TASK_NAME"
+
```
Comment (thomasrockhu) [inline codecov:844]:
```suggestion
  job="$CIRRUS_TASK_NAME"
```

File: codecov (hunk 2 above)
Comment (hootener) [reply codecov:820]: commit this suggestion for simple line cleanup

File: SHA1SUM [hunk 5]
```diff
@@ -1 +1 @@
-c03d34fc54383953b7a03ae7fcc6dec0990d76a7  codecov
+8fbba975dccd40ae06704bcea67b70bb3f8e919d  codecov
```
Comment (hootener) [inline SHA1SUM:1]: @wbeuil removal of trailing whitespace made shasum invalid. Do you mind regenerating? Once it passes in CI I'll merge.

Comment (thomasrockhu) [inline SHA1SUM:1]:
```suggestion
a80d06cabad50e1e9cbb2c01b6db59db12da1839. codecov
```

**Pull request merged**